specutils/io/default_loaders/wcs_fits.py

```diff
@@ -86,3 +91,406 @@ def wcs1d_fits_writer(spectrum, file_name, **kwargs):
+                                               key,
+                                               wat_wcs_dict[key]))
+
+    if 'spec1' in wat_wcs_dict.keys():
+        # print(wat_wcs_dict['spec1'])
+        spec = wat_wcs_dict['spec1'].split()
+        aperture = int(spec[0])
+        beam = int(spec[1])
+        disp_type = int(spec[2])
+        disp_start = float(spec[3])
+        disp_del_av = float(spec[4])
+        pix_num = int(spec[5])
+        dopp_fact = float(spec[6])
+        aper_low = int(float(spec[7]))
+        aper_high = int(float(spec[8]))
+        weight = float(spec[9])
+        zeropoint = float(spec[10])
+        function_type = int(spec[11])
+        order = int(float(spec[12]))
+        min_pix_val = int(float(spec[13]))
+        max_pix_val = int(float(spec[14]))
+        # resto = list(spec[9:])
+        params = [float(i) for i in spec[15:]]
+        # TODO (simon): Document wcs_dict (what is what).
+        wcs_dict = {'aperture': aperture,
+                    'beam': beam,
+                    'dtype': disp_type,
+                    'dstart': disp_start,
+                    'avdelt': disp_del_av,
+                    'pnum': pix_num,
+                    'z': dopp_fact,
+                    'alow': aper_low,
+                    'ahigh': aper_high,
+                    'weight': weight,
+                    'zeropoint': zeropoint,
+                    'ftype': function_type,
+                    'order': order,
+                    'pmin': min_pix_val,
+                    'pmax': max_pix_val,
+                    'fpar': params}
+
+        # This section of the code only shows a plot to see if the code
+        # above actually worked which means this methods has not been fully
+        # developed neither tested
+        logging.info('Retrieving model')
+        math_model = _set_math_model(wcs_dict=wcs_dict)
+
+        spectral_axis = math_model(range(1, wcs_dict['pnum'] + 1))
+        return spectral_axis
+
+
+def _set_math_model(wcs_dict):
+    """Defines a mathematical model
+
+    Uses 2 keywords to decided which model is to be built and calls the
+    appropriate function.
+
+    dtype:
+
+        -1 : None, no wavelength solution available
+        0 : Linear wavelength solution
+        1: Log-Linear wavelength solution (not implemented)
+        2: Non-Linear solutions
+            ftype:
+                1: Chebyshev
+                2: Legendre
+                3: Linear Spline (not implemented)
+                4: Cubic Spline (not implemented)
+                5: Pixel Coordinates (not implemented)
+
+    Not implemented models could be implemented on user-request.
+
+
+    Parameters
+    ----------
+
+    wcs_dict : dict
+        Contains all the necessary wcs information needed for building any of
+        models supported.
+
+
+    Returns
+    -------
+
+    `~astropy.modeling.Model`
+
+    """
+    if wcs_dict['dtype'] == -1:
+        return _none()
```

Comment:
Why is this a function call? Can you just add `raise NotImplementError` here instead?

Reply:
I think it should return the pixel axis. I will look it up later. I'm thinking right now on returning a model such as `f(x) = x`